jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -303,27 +449,67 @@ void updateBuckets(Set<V> vertices)
             /**
              * Set of vertices currently stored in this bucket.
              */
-            private Set<V> vertices;
+            private Queue<V> vertices;
```

Comment:
If you change the underlying data structure from HashSet to PriorityQueue, does the resulting iterator runs in linear time? When we retrieve a vertex from the bucket with the lexicographically largest label (from the head), we have to move all its neighbors to new buckets. This involves removing them from old buckets. Single remove operation runs in time proportional to the size of the priority queue, therefore the total running time is at least quadratic.

Reply:
Just for completeness' sake: You are absolutely correct. We fixed this issue now.

Thank you for taking the time to review our code and pointing out this mistake of mine!